src/engine/source/hlp/src/hlp.cpp

```diff
@@ -12,186 +12,225 @@
-                result[parser.name + ".folder"] = std::move(filePathResult.folder);
-                result[parser.name + ".name"] = std::move(filePathResult.name);
-                result[parser.name + ".extension"] = std::move(filePathResult.extension);
+        }
+    }
+    else
+    {
+        auto it = kECSParserMapper.find(parser.name);
+        if(it != kECSParserMapper.end())
+        {
+            parser.type = it->second;
+        }
+    }
+
+    setParserOptions(parser, args);
+
+    return parser;
+}
+
+std::vector<Parser> getParserList(ExpresionList const &expresions)
+{
+    std::vector<Parser> parsers;
+
+    for(auto const &expresion : expresions)
+    {
+        switch(expresion.type)
+        {
+            case ExpresionType::Capture:
+            case ExpresionType::OptionalCapture:
+            case ExpresionType::OrCapture:
+            {
+                parsers.push_back(createParserFromExpresion(expresion));
                 break;
             }
-            case ParserType::UserAgent:
+            case ExpresionType::Literal:
             {
-                UserAgentResult userAgent;
-                if(parseUserAgent(&eventIt,
-                                  parser.endToken,
-                                  userAgent))
-                {
-                    result[parser.name + ".original"] =
-                        std::move(userAgent.original);
-                }
+                Parser p;
+                p.name = expresion.text;
+                p.type = ParserType::Literal;
+                p.expType = ExpresionType::Literal;
+                p.endToken = expresion.endToken;
+                parsers.push_back(p);
                 break;
             }
-            default: {
-                fprintf(stderr,
-                        "Missing implementation for parser type: [%i]\n",
-                        parser.parserType);
+            default:
+            {
+                // TODO report error
                 break;
             }
         }
+    }
+
+    return parsers;
+}
+
+static bool executeParserList(std::string const &event,
+                              ParserList const &parsers,
+                              ParseResult &result)
+{
+    const char *eventIt = event.c_str();
+
+    // TODO This implementation is super simple for the POC
+    // but we will want to re-do it or revise it to implement
+    // better parser combinations
+    bool isOk = false;
+    for(auto const &parser : parsers)
+    {
+        const char *prevIt = eventIt;
+        auto parseFunc = kAvailableParsers[static_cast<int>(parser.type)];
+        if(parseFunc != nullptr)
+        {
+            isOk = parseFunc(&eventIt, parser, result);
+        }
+        else
+        {
+            isOk = false;
+            fprintf(stderr,
+                    "Missing implementation for parser type: [%i]\n",
+                    parser.type);
+            break;
```

Comment:
We shouldn´t break the loop, so the isOk check below take place.

Reply:
done